project-base/app/src/DataFixtures/Demo/NavigationItemDataFixture.php

```diff
@@ -106,20 +160,20 @@ public function getDependencies(): array
 
     /**
      * @param \Shopsys\FrameworkBundle\Model\Navigation\NavigationItemData $navigationItemData
+     * @param array $categoriesByColumn
      */
-    private function addCategoriesToNavigationItem(NavigationItemData $navigationItemData): void
-    {
-        $navigationItemData->categoriesByColumnNumber[1] = [
-            $this->getCategoryReference(CategoryDataFixture::CATEGORY_ELECTRONICS),
-            $this->getCategoryReference(CategoryDataFixture::CATEGORY_BOOKS),
-            $this->getCategoryReference(CategoryDataFixture::CATEGORY_TOYS),
-        ];
-        $navigationItemData->categoriesByColumnNumber[2] = [
-            $this->getCategoryReference(CategoryDataFixture::CATEGORY_GARDEN_TOOLS),
-        ];
-        $navigationItemData->categoriesByColumnNumber[3] = [
-            $this->getCategoryReference(CategoryDataFixture::CATEGORY_FOOD),
-        ];
+    private function addCategoriesToNavigationItem(
+        NavigationItemData $navigationItemData,
+        array $categoriesByColumn,
+    ): void {
+        foreach ($categoriesByColumn as $columnNumber => $categories) {
```

Comment:
...and this `$categoryReferenceNames`, but I do not insist on that, it just feels more understandable that way